flake.nix

```diff
@@ -0,0 +1,100 @@
+        "categorifier-hedgehog" =
+          self.callCabal2nix "categorifier-hedgehog" ./hedgehog { };
+        "categorifier-adjunctions-integration" =
+          self.callCabal2nix "categorifier-adjunctions-integration"
+          ./integrations/adjunctions/integration-test { };
+        "categorifier-adjunctions-integration-test" =
+          self.callCabal2nix "categorifier-adjunctions-integration-test"
+          ./integrations/adjunctions/integration { };
+        "categorifier-categories-integration" =
+          self.callCabal2nix "categorifier-categories-integration"
+          ./integrations/categories/integration { };
+        "categorifier-categories-integration-test" =
+          self.callCabal2nix "categorifier-categories-integration-test"
+          ./integrations/categories/integration-test { };
+        "categorifier-concat-examples" =
+          self.callCabal2nix "categorifier-concat-examples"
+          ./integrations/concat/examples { };
+        "categorifier-concat-integration" =
+          self.callCabal2nix "categorifier-concat-integration"
+          ./integrations/concat/integration { };
+        "categorifier-concat-integration-test" =
+          self.callCabal2nix "categorifier-concat-integration-test"
+          ./integrations/concat/integration-test { };
+        "categorifier-concat-extensions-category" =
+          self.callCabal2nix "categorifier-concat-extensions-category"
+          ./integrations/concat-extensions/category { };
+        "categorifier-concat-extensions-integration" =
+          self.callCabal2nix "categorifier-concat-extensions-integration"
+          ./integrations/concat-extensions/integration { };
+        "categorifier-concat-extensions-integration-test" =
+          self.callCabal2nix "categorifier-concat-extensions-integration-test"
+          ./integrations/concat-extensions/integration-test { };
+        "categorifier-unconcat-category" =
+          self.callCabal2nix "categorifier-unconcat-category"
+          ./integrations/unconcat/category { };
+        "categorifier-unconcat-integration" =
+          self.callCabal2nix "categorifier-unconcat-integration"
+          ./integrations/unconcat/integration { };
+        "categorifier-unconcat-integration-test" =
+          self.callCabal2nix "categorifier-unconcat-integration-test"
+          ./integrations/unconcat/integration-test { };
+        "categorifier-vec-integration" =
+          self.callCabal2nix "categorifier-vec-integration"
+          ./integrations/vec/integration { };
+        "categorifier-vec-integration-test" =
+          self.callCabal2nix "categorifier-vec-integration-test"
+          ./integrations/vec/integration-test { };
+        "categorifier-plugin" =
+          self.callCabal2nix "categorifier-plugin" ./plugin { };
+        "categorifier-plugin-test" =
+          self.callCabal2nix "categorifier-plugin-test" ./plugin-test { };
+        "categorifier-th" = self.callCabal2nix "categorifier-th" ./th { };
```

Comment:
I’m not a fan of this duplication of all the cabal files. Is there some way we can discover them, either via `find` or (even better) grabbing them from `cabal.project`?

Reply:
Yes. I am currently looking into a solution for this. Definitely, there are some examples in which the cabal.project file is parsed and packages are automatically included, but needs some significant coding in nix lang. I prefer to have that improvement in a separate PR if it turns out to be somewhat heavy solution.

Reply:
There’s some discussion at https://github.com/NixOS/cabal2nix/issues/286, which I don’t completely follow, other than that the same sort of functionality already exists for Stack and that perhaps this can be done via `shellFor`?

Certainly seems like `cabal2nix` is the right place for the code to live, in any case.

Reply:
oh. that's a great find. Yeah, if it is supported by official cabal2nix, that's awesome!

Reply:
From what I examined, the cabal2nix tooling seems not to fully support cabal.project yet.  
so as a temporary solution, currently in this PR, package sets defined in cabal.project is extracted minimally (quite easy to be broken) through nix expression. As @sellout suggested, the cabal2nix discussion is referred and we will continue to follow or to contribute to that.